<hr>

# 📊 DATA EXPLORATION 📊

<style>
h1 {
    text-align: center;
    color: green;
    font-weight: bold;
}
</style>

<style>
h2 {
    text-align: center;
    color: black;
    font-weight: bold;
}
</style>

<style>
h3 {
    text-align: center;
    color: black;
    font-weight: bold;
}
</style>

<style>
h4 {
    text-align: center;
    color: black;
    font-weight: bold;
}
</style>

<hr>

In [30]:
import pandas as pd
df = pd.read_csv("../data/raw/online_shoppers_intention.csv")

In [31]:
from pathlib import Path

descr_path = Path("..") / "data" / "metadata" / "online_shoppers_DESCR.txt"
DESCR = descr_path.read_text(encoding="utf-8")

print(DESCR)

Online Shoppers Purchasing Intention Dataset

Description
-----------
This dataset contains session-level data from an e-commerce website.
Each row represents a unique user session.

The dataset consists of:
- 10 numerical attributes
- 8 categorical attributes
- 1 binary target variable ("Revenue")

The goal is to predict whether a user session will result in a completed transaction.

Target Variable
---------------
Revenue:
    TRUE  -> The session ended with a purchase.
    FALSE -> No purchase was completed.

Feature Descriptions
--------------------

Administrative:
    Number of administrative pages (any non-product page like login, account settings, shipping pages) visited during the session.

Administrative_Duration:
    Total time spent on administrative pages (in seconds).

Informational:
    Number of informational pages (any non-product info page like About Us, FAQ, Policy pages) visited during the session.

Informational_Duration:
    Total time spent on informational pages

In [32]:
df.shape

(12330, 18)

In [33]:
df.dtypes

Administrative               int64
Administrative_Duration    float64
Informational                int64
Informational_Duration     float64
ProductRelated               int64
ProductRelated_Duration    float64
BounceRates                float64
ExitRates                  float64
PageValues                 float64
SpecialDay                 float64
Month                       object
OperatingSystems             int64
Browser                      int64
Region                       int64
TrafficType                  int64
VisitorType                 object
Weekend                       bool
Revenue                       bool
dtype: object

In [34]:
df.isna().sum()

Administrative             0
Administrative_Duration    0
Informational              0
Informational_Duration     0
ProductRelated             0
ProductRelated_Duration    0
BounceRates                0
ExitRates                  0
PageValues                 0
SpecialDay                 0
Month                      0
OperatingSystems           0
Browser                    0
Region                     0
TrafficType                0
VisitorType                0
Weekend                    0
Revenue                    0
dtype: int64

- No null values, suitable for a classification ML model

In [35]:
df["Revenue"].value_counts(normalize=True).round(4) * 100

Revenue
False    84.53
True     15.47
Name: proportion, dtype: float64

- There is a strong class imbalance with 84.53% negative conversion (no purchase) vs 15.47% sessions that ended in a purchase

In [36]:
df["VisitorType"].value_counts()

VisitorType
Returning_Visitor    10551
New_Visitor           1694
Other                   85
Name: count, dtype: int64